atspi-common/src/operation.rs

```diff
@@ -0,0 +1,33 @@
+/// An operation can either be [`Self::Insert`] or [`Self::Delete`].
+/// These correspond to methods available on [`Vec`].
+#[derive(Debug, PartialEq, Clone, serde::Serialize, serde::Deserialize, Eq, Hash, Default)]
+pub enum Operation {
+	#[default]
+	#[serde(rename = "add")]
+	#[serde(alias = "add/system")]
+	Insert,
+	#[serde(rename = "delete")]
+	#[serde(alias = "delete/system")]
+	Delete,
+}
+
+impl TryFrom<&str> for Operation {
```

Comment:
stdlib also offers `FromStr` which, if implemented, allows you to do:

```Rust
        let x = "add";
        let x_fromstr = x.parse::<Operation>().map_err(|_| CreateError)?;
```
It may come down to taste, so I'll leave it up to you.